### DISCUSSION TITLE
Unable to connect: x509: certificate signed by unknown authority

### DISCUSSION THREAD

**Discussion by israaaa270** [ory/hydra#2336] [Q&A] [answered]
I run binary hydra from command line 

  -`hydra clients create --endpoint https://127.0.0.1:4445/` 
 Unable to connect: x509: certificate signed by unknown authority

- in docker hydra 
   2021/02/02 04:45:24 http: TLS handshake error from 127.0.0.1:53248: remote error: tls: bad certificate

how I solve this?

**Comment by vinckr** [comment, marked as answer]
This is probably some network / Docker problem? 
What is port 53248? 

Please review your Docker setup, maybe clean it all up if possible or test in a new enviroment.

**Reply by israaaa270** [reply]
What is port 53248?
is generated random as ,
2021/02/02 12:26:12 http: TLS handshake error from 127.0.0.1:57985: remote error: tls: bad certificate
2021/02/02 12:26:13 http: TLS handshake error from 127.0.0.1:57986: remote error: tls: bad certificate
2021/02/02 12:26:13 http: TLS handshake error from 127.0.0.1:57987: remote error: tls: bad certificate
2021/02/02 12:26:14 http: TLS handshake error from 127.0.0.1:57988: remote error: tls: bad certificate
2021/02/02 12:26:15 http: TLS handshake error from 127.0.0.1:57989: remote error: tls: bad certificate
2021/02/02 12:26:16 http: TLS handshake error from 127.0.0.1:57990: remote error: tls: bad certificate
2021/02/02 12:26:17 http: TLS handshake error from 127.0.0.1:57991: remote error: tls: bad certificate
2021/02/02 12:26:17 http: TLS handshake error from 127.0.0.1:57992: remote error: tls: bad certificate
2021/02/02 12:26:19 http: TLS handshake error from 127.0.0.1:57993: remote error: tls: bad certificate
2021/02/02 12:26:20 http: TLS handshake error from 127.0.0.1:57994: remote error: tls: bad certificate

in docker that port not exist.
only port :4444,4445 exist in docker.

**Reply by vinckr** [reply]
You need to use `--skip-tls-verify` because you are using a self signed certificate. 
If you google `tls: bad certificate golang`, there are resources that explain it - [for example](https://forum.golangbridge.org/t/tls-handshake-error/6125/4)

**Reply by israaaa270** [reply]
where can I put `--skip-tls-verify`   , the command `hydra serve `  doesn't support this flag.

**Reply by vinckr** [reply]
Yea, it is a flag for `hydra clients` or `hydra token`. 
You skip the tls verification for that client or token.
More info on the CLI commands here: https://www.ory.sh/hydra/docs/cli/
More info on the flow and the flag here: https://www.ory.sh/hydra/docs/configure-deploy/#perform-oauth-20-flow

**Reply by israaaa270** [reply]
thanks so much @vinckr 🥇
